cmd/cosign/cli/verify/verify.go

```diff
@@ -186,9 +189,32 @@ func (c *VerifyCommand) Exec(ctx context.Context, images []string) (err error) {
 					co.IntermediateCerts.AddCert(cert)
 				}
 			}
-		} else {
-			// This performs an online fetch of the Fulcio roots. This is needed
-			// for verifying keyless certificates (both online and offline).
+		case c.CARoots != "":
+			caRoots, err := loadCertChainFromFileOrURL(c.CARoots)
+			if err != nil {
+				return err
+			}
+			co.RootCerts = x509.NewCertPool()
+			if len(caRoots) > 0 {
```

Comment:
nit: should we error out if `len(caRoots)` is `0`?

Reply:
is it a "fatal" condition?  Even if `x509.NewCertPool()` returns no roots, there should can be some added to `co.RootCerts` in the first line of this function.